### PR TITLE
Add storage defender support

### PR DESCRIPTION
**Pull request by shanoor**
## PR Checklist

---

- [ ] I have added example(s) inside the [./examples/] folder
- [ ] I have added the example(s) to the integration test list for [normal (~30 minutes)](./workflows/standalone-scenarios.json) or [long runner >30 minutes](./workflows/standalone-scenarios-longrunners.json)
- [x] I have checked the [coding conventions as per the wiki](https://github.com/aztfmod/terraform-azurerm-caf/wiki)
- [x] I have checked to ensure there aren't other open Pull Requests for the same update/change?

## Description

Add support for enabling Defender for Storage on a storage account.

## Does this introduce a breaking change

- [ ] YES
- [x] NO

### CODE REVIEW

File: modules/storage_account/storage_defender.tf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+resource "azurerm_security_center_storage_defender" "defender" {
+  count = can(var.storage_account.defender) ? 1 : 0
+
+  storage_account_id                          = azurerm_storage_account.stg.id
+  override_subscription_settings_enabled      = try(var.storage_account.defender.override_subscription_settings, false)
```
Comment (arnaudlh) [inline modules/storage_account/storage_defender.tf:5]:
```suggestion
  override_subscription_settings_enabled      = try(var.storage_account.defender.override_subscription_settings, null)
```

File: modules/storage_account/storage_defender.tf [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  malware_scanning_on_upload_enabled          = try(var.storage_account.defender.malware_scanning_on_upload, false)
```
Comment (arnaudlh) [inline modules/storage_account/storage_defender.tf:6]:
```suggestion
  malware_scanning_on_upload_enabled          = try(var.storage_account.defender.malware_scanning_on_upload, null)
```

File: modules/storage_account/storage_defender.tf [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+  malware_scanning_on_upload_cap_gb_per_month = try(var.storage_account.defender.malware_scanning_on_upload_cap_gb_per_month, -1)
```
Comment (arnaudlh) [inline modules/storage_account/storage_defender.tf:7]:
```suggestion
  malware_scanning_on_upload_cap_gb_per_month = try(var.storage_account.defender.malware_scanning_on_upload_cap_gb_per_month, null)
```

File: modules/storage_account/storage_defender.tf [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+  sensitive_data_discovery_enabled            = try(var.storage_account.defender.sensitive_data_discovery_enabled, false)
```
Comment (arnaudlh) [inline modules/storage_account/storage_defender.tf:8]:
```suggestion
  sensitive_data_discovery_enabled            = try(var.storage_account.defender.sensitive_data_discovery_enabled, null)
```

**Review by arnaudlh — CHANGES_REQUESTED** [review]
Thanks for the PR @shanoor, are you able to change it to default the values to the provider instead of the module? 
Can you please also add an example and the CI/CD line?

**Comment by shanoor** [comment]
> Thanks for the PR @shanoor, are you able to change it to default the values to the provider instead of the module? Can you please also add an example and the CI/CD line?

I committed your suggestions and added an example.

**Review by arnaudlh — APPROVED** [review]
lgtm

**Pull request merged**